internal/pkg/instrumentors/utils/kernel.go

```diff
@@ -15,28 +15,39 @@
 package utils
 
 import (
-	"strings"
+	"fmt"
+	"strconv"
 	"syscall"
 
 	"github.com/hashicorp/go-version"
 )
 
+// parse logic adapted from https://github.com/golang/go/blob/go1.21.3/src/internal/syscall/unix/kernel_version_linux.go
 func GetLinuxKernelVersion() (*version.Version, error) {
-	var utsname syscall.Utsname
-
-	if err := syscall.Uname(&utsname); err != nil {
+	var uname syscall.Utsname
+	if err := syscall.Uname(&uname); err != nil {
 		return nil, err
 	}
 
-	var buf [65]byte
-	for i, v := range utsname.Release {
-		buf[i] = byte(v)
-	}
-
-	ver := string(buf[:])
-	if strings.Contains(ver, "-") {
-		ver = strings.Split(ver, "-")[0]
+	var (
+		values    [2]int
+		value, vi int
+	)
+	for _, c := range uname.Release {
+		if '0' <= c && c <= '9' {
+			value = (value * 10) + int(c-'0')
+		} else {
+			// Note that we're assuming N.N.N here.
+			// If we see anything else, we are likely to mis-parse it.
+			values[vi] = value
+			vi++
+			if vi >= len(values) {
+				break
+			}
+			value = 0
+		}
 	}
+	ver := fmt.Sprintf("%s.%s", strconv.Itoa(values[0]), strconv.Itoa(values[1]))
```

Comment:
This doesn't seem right, shouldn't this be of the form "X.X.X" to match the existing behavior?

Reply:
Fix later.

Reply:
This is the use we are doing in this function:
```
ver, err := utils.GetLinuxKernelVersion()
...

minVersion := version.Must(version.NewVersion("5.14"))
log.Logger.V(0).Info("Detected linux kernel version", "version", ver)
if ver.GreaterThanOrEqual(minVersion) {
advice = syscall.MADV_WILLNEED | MadvisePopulateRead | MadvisePopulateWrite
}
```

Since we are comparing to 5.14 I think it should be enough to take two segments of the version.
In addition, I don't see a reason why not to call https://pkg.go.dev/internal/syscall/unix#KernelVersion directly.
Also this might be worth considering: https://pkg.go.dev/github.com/moby/moby/pkg/parsers/kernel

Reply:
@RonFed 
> Since we are comparing to 5.14 I think it should be enough to take two segments of the version.

I thought too.

> In addition, I don't see a reason why not to call https://pkg.go.dev/internal/syscall/unix#KernelVersion directly

That package path is `internal/syscall/unix`, we can't call cuz have `internal`, therefore I copied.

> Also this might be worth considering: https://pkg.go.dev/github.com/moby/moby/pkg/parsers/kernel

Interesting, I'll check it tomorrow.

Reply:
@zchee Any updates ?

Reply:
@RonFed Sorry, what do you want any change? 

> Since we are comparing to 5.14 I think it should be enough to take two segments of the version.

Current change will be get `6.16` or etc. Should I change to get `X.X.X` as @MrAlias said?

Reply:
I think it's Ok.